torch_xla/experimental/tpu.py

```diff
@@ -94,14 +109,15 @@ def version() -> int:
   except requests.HTTPError as e:
     raise EnvironmentError('Failed to get TPU metadata') from e
 
-  match = re.match(r'^v(\d)-(\d+)$', env['ACCELERATOR_TYPE'])
+  match = re.match(r'^v(\d)-(\d+)$', env[xenv.ACCELERATOR_TYPE])
   return int(match.groups()[0])
 
 
 def get_worker_ips() -> List[str]:
   """Returns ordered list of TPU worker IPs from TPU metadata."""
-  metadata = _get_metadata('worker-network-endpoints')
-
+  metadata = xu.getenv_as(xenv.TPU_WORKER_HOSTNAMES, str, '')
```

Comment:
TPU_WORKER_HOSTNAMES is of the format: host-name-1,host-name-2,..........,host-name-n

Change the code to the following perhaps?
  if metadata is '':
    metadata = _get_metadata('worker-network-endpoints')
    # Workers have format 'hostname:uid:ip,hostname:uid:ip,...'
    workers = metadata.split(',')
    ips = [worker.split(':')[2] for worker in workers]
  else:
    ips = metadata.split(',')
  return ips if len(ips) > 1 else ['localhost']

Reply:
Done